iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/SecretsGenerationCheck.java

```diff
@@ -62,20 +66,164 @@ public class SecretsGenerationCheck implements IacCheck {
+
+  private static final CommandDetector SSHPASS_P_FLAG_NO_SPACE_PWD = CommandDetector.builder()
+    .with("sshpass")
+    .withAnyExcludingIncludeUnresolved(arg -> !arg.startsWith("-p"))
+    .withArgumentResolutionIncludeUnresolved(resolution -> resolution.value().startsWith("-p") &&
+      (resolution.value().length() > 2 || resolution.argument().expressions().size() > 1))
+    .build();
+
+  private static CommandDetector wgetFlagEquals(String flag) {
+    String flagAndEquals = flag + "=";
+    return CommandDetector.builder()
+      .with("wget")
+      .withAnyExcludingIncludeUnresolved(arg -> !arg.startsWith(flagAndEquals))
+      .withIncludeUnresolved(arg -> arg.startsWith(flagAndEquals))
+      .build();
+  }
+
+  private static CommandDetector commandFlagSpace(String command, String flag) {
+    return CommandDetector.builder()
+      .with(command)
+      .withOptionalRepeatingExcept(flag)
+      .with(flag::equals)
+      .withIncludeUnresolved(a -> true)
+      .build();
+  }
+
   private static final Set<CommandDetector> DETECTORS = Set.of(
     SSH_DETECTOR,
     KEYTOOL_DETECTOR,
     SENSITIVE_OPENSSL_COMMANDS);
 
+  private static final Set<CommandDetector> DETECTORS_IGNORE_MOUNT_SECRET = Set.of(
+    WGET_PASSWORD_FLAG_EQUALS_PWD,
+    WGET_PASSWORD_FLAG_SPACE_PWD,
+    WGET_FTP_PASSWORD_FLAG_EQUALS_PWD,
+    WGET_FTP_PASSWORD_FLAG_SPACE_PWD,
+    WGET_HTTP_PASSWORD_FLAG_EQUALS_PWD,
+    WGET_HTTP_PASSWORD_FLAG_SPACE_PWD,
+    WGET_PROXY_PASSWORD_FLAG_EQUALS_PWD,
+    WGET_PROXY_PASSWORD_FLAG_SPACE_PWD,
+    SSHPASS_P_FLAG_NO_SPACE_PWD,
+    SSHPASS_P_FLAG_SPACE_PWD);
+
+  private static final Set<CommandDetector> CURL_DETECTORS = Set.of(CURL_USER_FLAG_SPACE_PWD,
+    CURL_USER_SHORT_FLAG_SPACE_PWD);
+
   @Override
   public void initialize(InitContext init) {
-    init.register(RunInstruction.class, SecretsGenerationCheck::checkRunInstruction);
+    init.register(DockerImage.class, SecretsGenerationCheck::checkDockerImage);
+  }
+
+  private static void checkDockerImage(CheckContext ctx, DockerImage dockerImage) {
+    if (!dockerImage.isLastDockerImageInFile()) {
+      return;
+    }
+
+    List<RunInstruction> runInstructions = dockerImage.instructions().stream()
+      .filter(RunInstruction.class::isInstance)
+      .map(RunInstruction.class::cast)
+      .collect(Collectors.toList());
+
+    runInstructions.forEach(runInstruction -> checkRunInstruction(ctx, runInstruction));
   }
 
   private static void checkRunInstruction(CheckContext ctx, RunInstruction runInstruction) {
     List<ArgumentResolution> resolvedArgument = CheckUtils.resolveInstructionArguments(runInstruction);
 
     DETECTORS.forEach(
       detector -> detector.search(resolvedArgument).forEach(command -> ctx.reportIssue(command, MESSAGE)));
+
+    if (isMountSecret(runInstruction)) {
+      return;
+    }
+
+    DETECTORS_IGNORE_MOUNT_SECRET.forEach(
```

Comment:
I would not add the `IGNORE_MOUNT_SECRET` to this name, as the detectors don't actively ignore the mount secret. It's only ignored because we are after the if call.
I would add a comment above this line:
`The detectors aren't evaluating a possible mount secret flag, because it is handled in the if condition above`

Reply:
> I would not add the IGNORE_MOUNT_SECRET
But then I will have `DETECTORS` list 😢 
Let's rename in that way (to describe what those detectors do):
`DETECTORS` -> `DETECTORS_THAT_STORE_SECRETS`
`DETECTORS_IGNORE_MOUNT_SECRET ` -> `DETECTORS_THAT_HAVE_SECRETS_IN_CMD`

Reply:
Also added a different comment with an explanation.